diff -Nur quickshell-0.2.0.og/src/CMakeLists.txt quickshell-0.2.0/src/CMakeLists.txt
--- quickshell-0.2.0.og/src/CMakeLists.txt	2025-07-27 06:50:52.000000000 +0100
+++ quickshell-0.2.0/src/CMakeLists.txt	2025-09-28 18:16:24.685237798 +0100
@@ -1,3 +1,4 @@
+find_package(XKB REQUIRED)
 qt_add_executable(quickshell main.cpp)
 
 install(TARGETS quickshell RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
diff -Nur quickshell-0.2.0.og/src/wayland/CMakeLists.txt quickshell-0.2.0/src/wayland/CMakeLists.txt
--- quickshell-0.2.0.og/src/wayland/CMakeLists.txt	2025-07-27 06:50:52.000000000 +0100
+++ quickshell-0.2.0/src/wayland/CMakeLists.txt	2025-09-28 18:16:43.329673304 +0100
@@ -1,5 +1,6 @@
 find_package(PkgConfig REQUIRED)
 find_package(WaylandScanner REQUIRED)
+find_package(Qt6 COMPONENTS WaylandClient WaylandClientPrivate REQUIRED)
 pkg_check_modules(wayland REQUIRED IMPORTED_TARGET wayland-client wayland-protocols)
 
 # wayland protocols
diff -Nur quickshell-0.2.0.og/src/window/CMakeLists.txt quickshell-0.2.0/src/window/CMakeLists.txt
--- quickshell-0.2.0.og/src/window/CMakeLists.txt	2025-07-27 06:50:52.000000000 +0100
+++ quickshell-0.2.0/src/window/CMakeLists.txt	2025-09-28 18:14:56.744377423 +0100
@@ -1,3 +1,4 @@
+find_package(Qt6 COMPONENTS Quick QuickPrivate REQUIRED)
 qt_add_library(quickshell-window STATIC
 	proxywindow.cpp
 	windowinterface.cpp
